build/jemalloc.m4

```diff
@@ -68,7 +68,18 @@ if test "$enable_jemalloc" != "no"; then
     AC_CHECK_HEADERS(jemalloc/jemalloc.h, [jemalloc_have_headers=1])
   fi
   if test "$jemalloc_have_headers" != "0"; then
-    jemalloch=1
+    AC_RUN_IFELSE([
+      AC_LANG_PROGRAM(
+        [#include <jemalloc/jemalloc.h>],
+        [
+          #if (JEMALLOC_VERSION_MAJOR == 0)
+          exit(1);
+          #endif
```

Comment:
Will this just exit without an error message during the configure phase?

Reply:
It will say this when running configure if jemalloc has a bogus version:
`configure: error: jemalloc has bogus version`

Reply:
Ah, I see.

Reply:
Will ./configure have a non-zero status code?

Reply:
yes it will